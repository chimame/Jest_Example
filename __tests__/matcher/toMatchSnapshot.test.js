import React from 'react'

const sample = (renderChild) => {
  let child = null
  if (renderChild) {
    child = <span>aaa</span>
  }

  return (
    <div className={'pearent'}>
      {child}
    </div>
  )
}

describe('toMatchSnapshot example', () => {
  test('render child', () => {
    expect(sample(true)).toMatchSnapshot()
  })

  test('not render child', () => {
    expect(sample(false)).toMatchSnapshot()
  })
})
