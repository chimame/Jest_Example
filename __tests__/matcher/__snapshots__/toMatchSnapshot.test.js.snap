exports[`toMatchSnapshot example not render child 1`] = `
<div
  className="pearent" />
`;

exports[`toMatchSnapshot example render child 1`] = `
<div
  className="pearent">
  <span>
    aaa
  </span>
</div>
`;
